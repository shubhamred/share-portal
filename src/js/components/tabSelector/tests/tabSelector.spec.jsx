import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import TabSelector from '../tabSelector';

configure({ adapter: new Adapter() });

describe('components/Tab Selctor', () => {
  it('Renders default options', () => {
    const props = {
      menuItems: []
    };
    const component = shallow(
      // eslint-disable-next-line react/jsx-props-no-spreading
      <TabSelector {...props} />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('Renders with Props', () => {
    const props = {
      menuItems: [{ id: 'dummyId1', name: 'Dummy 1' }, { id: 'dummyId2', name: 'Dummy 2' }],
      selectedTab: 'dummyId1'
    };
    const component = shallow(
      // eslint-disable-next-line react/jsx-props-no-spreading
      <TabSelector {...props} />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('Components hadles On click', () => {
    const props = {
      menuItems: [{ id: 'dummyId1', name: 'Dummy 1' }, { id: 'dummyId2', name: 'Dummy 2' }],
      selectedTab: 'dummyId1',
      onChangeTab: jest.fn()
    };
    const component = shallow(
      // eslint-disable-next-line react/jsx-props-no-spreading
      <TabSelector {...props} />
    );
    component.findWhere((node) => node.key() === 'dummyId1').children().simulate('click');
    expect(props.onChangeTab).toHaveBeenCalledTimes(1);
  });
});
