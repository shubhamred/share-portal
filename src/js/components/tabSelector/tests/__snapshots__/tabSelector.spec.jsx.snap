// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Tab Selctor Renders default options 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  style={
    Object {
      "marginBottom": "30px",
    }
  }
  wrap="nowrap"
/>
`;

exports[`components/Tab Selctor Renders with Props 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  style={
    Object {
      "marginBottom": "30px",
    }
  }
  wrap="nowrap"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    item={true}
    key="dummyId1"
    xs={2}
  >
    <WithStyles(ForwardRef(ButtonBase))
      className="selectedTab"
      onClick={[Function]}
      styles="selectedTab"
    >
      Dummy 1
    </WithStyles(ForwardRef(ButtonBase))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    item={true}
    key="dummyId2"
    xs={2}
  >
    <WithStyles(ForwardRef(ButtonBase))
      className="normalTab"
      onClick={[Function]}
      styles="selectedTab"
    >
      Dummy 2
    </WithStyles(ForwardRef(ButtonBase))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
